docs/src/main/docs/guides/07_datasource.adoc

```diff
@@ -0,0 +1,345 @@
+`test` data source is requested.
+
+<3> The `testDataSource` field here, whose name is arbitrary, is typed
+with
+https://docs.oracle.com/javase/8/docs/api/javax/sql/DataSource.html[`DataSource`].
+Its protection level, `private` in this case, is immaterial, following
+CDI rules.  Helidon MP's CDI container will use the configuration
+described elsewhere in this document to create a new or retrieve an
+existing `DataSource` implementation instance whose name is specified
+by the `@Named` annotation, and will set this field's value to it.
+
+=== Coupling Your Application Code to a Particular Connection Pool
+
+If for some reason you wish to couple your application tightly to a
+particular connection pool implementation, you can choose a different
+data type for the field receiving an injected `DataSource`.
+
+You might choose to do this to take advantage of additional methods
+offered by either the Hikari connection pool classes or the Oracle
+Universal Connection Pool classes.
+
+In general, if you have no need for methods that are not present in
+the
+https://docs.oracle.com/javase/8/docs/api/javax/sql/DataSource.html[`javax.sql.DataSource`
+interface], tight coupling to a connection pool implementation is
+discouraged.
+
+==== Coupling Your Application Code to the Hikari Connection Pool
+
+For example, if you wish to couple your application tightly to the
+Hikari connection pool implementation of the
+https://docs.oracle.com/javase/8/docs/api/javax/sql/DataSource.html[`javax.sql.DataSource`
+interface], you can do this:
+
+[source,java]
+----
+import com.zaxxer.hikari.HikariDataSource; <1>
+
+@Inject
+@Named("test")
+private HikariDataSource testDataSource; <2>
+----
+
+<1> Your application will now require the Hikari connection pool
+classes on its compile and runtime classpaths.
+
+<2> The CDI container will use the Hikari connection pool
+configuration described elsewhere in this document to create a new or
+retrieve an existing `HikariDataSource` implementation instance whose
+name is specified by the `@Named` annotation, and will set this
+field's value to it.
+
+==== Coupling Your Application Code to the Oracle Universal Connection Pool
+
+If, instead, you wish to couple your application tightly to Oracle
+Universal Connection Pool classes and interfaces, you can do this:
+
+[source,java]
+----
+import oracle.ucp.jdbc.PoolDataSource; <1>
+
+@Inject
+@Named("test")
+private PoolDataSource testDataSource; <2>
+----
+
+<1> Your application will now require the Oracle Universal Connection
+Pool classes on its compile and runtime classpaths.
+
+<2> The CDI container will use the Oracle Universal Connection Pool
+configuration described elsewhere in this document to create a new or
+retrieve an existing
+https://docs.oracle.com/en/database/oracle/oracle-database/19/jjuar/oracle/ucp/jdbc/PoolDataSource.html[`PoolDataSource`]
+implementation instance whose name is specified by the `@Named`
+annotation, and will set this field's value to it.
+
```

Comment:
Do we have some examples to point to? Like:

For some complete examples see:

* [Hikari wtih MySQL](https://github.com/oracle/helidon/tree/{helidon-version}/examples/integrations/cdi/datasource-hikaricp-mysql)
* [Hikari with H2](https://github.com/oracle/helidon/tree/{helidon-version}/examples/integrations/cdi/datasource-hikaricp-h2)

Reply:
Hmm; we do, but they're kind of complicated.  Given that lines 342-344 above are already commenting on an example do you think I should link to them?  I'm inclined not to.  More thoughts welcome.

Reply:
I like the idea of pointing the user to a working example (or examples) that they can easily build and run.  If the above are too complicated (I'm not sure they are), then maybe a new example using UCP?

Reply:
Added some links to examples.

Reply:
@barchetta Does this look OK?